### ISSUE TITLE
[questsion] how to differ gl_FragCoord is redeclared or not redeclared ?

### ISSUE THREAD

**Issue by ZhiqianXia** [KhronosGroup/glslang#2901]
Hi , 
   I met a cts test, which need report a error for gl_FragCoord.  Because the specs says:
_If gl_FragCoord is redeclared in any fragment shader in a program, it must be **redeclared** in all the
fragment shaders in that program that have a static use gl_FragCoord._

In the compile stage , we can't report the error for two shader. So we need put the error at link-time. 
But glslang didn't differ the situation : gl_FragCoord is redeclared or not redeclared ?  So the linker
can't get the info. my question is : can we add a flag for gl_FragCoord to express the redeclared info ?
   

**shader 1: gl_FragCoord  redeclaration without_layout qualifer**

```
#extension GL_ARB_explicit_attrib_location : enable
#ifdef GL_ES
precision mediump float;
#endif

in vec4 gl_FragCoord;  // without layout qualifer
layout (location = 0) out vec4 myColor;

void ModifyBlue() {
  if (gl_FragCoord.y >= 10) {
    myColor.b = 0.8;
  }
}
```

**shader 2: no redeclaration but used**

```
#extension GL_ARB_explicit_attrib_location : enable
#ifdef GL_ES
precision mediump float;
#endif

in vec4 i;
layout (location = 0) out vec4 myColor;
const float eps=0.001;
void ModifyBlue();
void main() {
    myColor = vec4(0.2);
    ModifyBlue();
    if (gl_FragCoord.y == trunc(gl_FragCoord.y)) {   // no redeclaration but used
        myColor.g = 0.8;
    }
    if (gl_FragCoord.x == trunc(gl_FragCoord.x)) {
        myColor.r = 0.8;
    }
    vec4 diff = gl_FragCoord - i;
    if (abs(diff.z)>eps) myColor.b = 0.5;
    if (abs(diff.w)>eps) myColor.a = 0.5;
}
```

**Comment by ZhiqianXia** [comment]
fix the issue with https://github.com/KhronosGroup/glslang/pull/2902

**Comment by greg-lunarg** [comment]
Fixed with #2902
